chips/stm32f4xx/src/dma/mod.rs

```diff
@@ -776,774 +767,61 @@ enum TransferMode {
-            }
-        });
-    }
-
-    fn stream_set_data_width(&self, msize: Msize, psize: Psize) {
-        match self.streamid {
-            StreamId::Stream0 => {
-                self.dma1
-                    .registers
-                    .s0cr
-                    .modify(S0CR::PSIZE.val(psize.0 as u32));
-                self.dma1
-                    .registers
-                    .s0cr
-                    .modify(S0CR::MSIZE.val(msize.0 as u32));
-            }
-            StreamId::Stream1 => {
-                self.dma1
-                    .registers
-                    .s1cr
-                    .modify(S1CR::PSIZE.val(psize.0 as u32));
-                self.dma1
-                    .registers
-                    .s1cr
-                    .modify(S1CR::MSIZE.val(msize.0 as u32));
-            }
-            StreamId::Stream2 => {
-                self.dma1
-                    .registers
-                    .s2cr
-                    .modify(S2CR::PSIZE.val(psize.0 as u32));
-                self.dma1
-                    .registers
-                    .s2cr
-                    .modify(S2CR::MSIZE.val(msize.0 as u32));
-            }
-            StreamId::Stream3 => {
-                self.dma1
-                    .registers
-                    .s3cr
-                    .modify(S3CR::PSIZE.val(psize.0 as u32));
-                self.dma1
-                    .registers
-                    .s3cr
-                    .modify(S3CR::MSIZE.val(msize.0 as u32));
-            }
-            StreamId::Stream4 => {
-                self.dma1
-                    .registers
-                    .s4cr
-                    .modify(S4CR::PSIZE.val(psize.0 as u32));
-                self.dma1
-                    .registers
-                    .s4cr
-                    .modify(S4CR::MSIZE.val(msize.0 as u32));
-            }
-            StreamId::Stream5 => {
-                self.dma1
-                    .registers
-                    .s5cr
-                    .modify(S5CR::PSIZE.val(psize.0 as u32));
-                self.dma1
-                    .registers
-                    .s5cr
-                    .modify(S5CR::MSIZE.val(msize.0 as u32));
-            }
-            StreamId::Stream6 => {
-                self.dma1
-                    .registers
-                    .s6cr
-                    .modify(S6CR::PSIZE.val(psize.0 as u32));
-                self.dma1
-                    .registers
-                    .s6cr
-                    .modify(S6CR::MSIZE.val(msize.0 as u32));
-            }
-            StreamId::Stream7 => {
-                self.dma1
-                    .registers
-                    .s7cr
-                    .modify(S7CR::PSIZE.val(psize.0 as u32));
-                self.dma1
-                    .registers
-                    .s7cr
-                    .modify(S7CR::MSIZE.val(msize.0 as u32));
-            }
-        }
+impl<'a> From<&'a dma1::Stream<'static>> for Stream<'a> {
```

Comment:
This unfortunately cannot be used by the Usart module yet, as it would need a const From implementation.

TODO: Remove it?

Reply:
Sorry, why would this need a const From implementation?

Reply:
Ah, whoops, I mixed that up. I meant the impl for DmaPeripheral
``` rust
impl From<dma1::Dma1Peripheral> for DmaPeripheral {
```

I need to create a DmaPeripheral in the const `new_usartX` functions in `usart.rs`, which would be much simpler if I could do:
``` rust
dma1::Dma1Peripheral::USART3_TX.into(),
```
instead of the current:
``` rust
dma::DmaPeripheral::Dma1Peripheral(dma1::Dma1Peripheral::USART3_TX),
```
But that doesn't work, because the From<> impl doesn't work in `const` functions.

Reply:
This problem has been removed in the newest round of refactoring.